api/src/beacons/beacons.service.ts

```diff
@@ -37,18 +37,24 @@ export class BeaconsService {
     latitude,
     longitude,
     maxDistance,
+    tag,
   }: FindBeaconsDto): Promise<Beacon[]> {
     return this.beaconModel
```

Comment:
I think all you had to do was add `tag: tag` to this query after the constraint on `location`.